File: 0167-two-sum-ii-input-array-is-sorted/0167-two-sum-ii-input-array-is-sorted.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
        int i =0;
        int j = nums.length -1;
        int sum = 0;
        int[] res = new int[2];
        while(i<j){
            sum = nums[i] + nums[j];
            if(sum == target){
                res[0] = i+1;
                res[1] = j+1;
                return res;
            }
            else if (sum < target){
                i++;
            }
            else {
                j--;
            }
        }
        return res;
    }
}

File: 0167-two-sum-ii-input-array-is-sorted/0167-two-sum-ii-input-array-is-sorted.scala
object Solution {
    def twoSum(nums: Array[Int], target: Int): Array[Int] = {
        var i = 0
        var j = nums.length - 1
        var sum = 0
        var res = Array(0, 0)

        while (i < j) {
          sum = nums(i) + nums(j)
          if (sum == target) {
            res(0) = i + 1
            res(1) = j + 1
            return res
          } else if (sum < target) {
            i += 1
          } else {
            j -= 1
          }
        }
        res
    }
}